docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+The the community owner node MUST derive a magnet link from the newly created `WakuMessageArchiveIndex` so it can be distributed to community member nodes.
+
+## Message archive distribution
+
+Message archives are available via the BitTorrent network as soon as magnet links for them have been created.
+Other community member nodes will download the message archives from the BitTorrent network once they receive a magnet link that contains a message archive index.
+
+The community owner node MUST send magnet links containing message archive indices to a special community channel. The topic of that special channel consists of the community id followed by the string `-archives`:
+
+```
+{community_id}-archives
+```
+
+All messages sent with this topic MUST be instances of `ApplicationMetadataMessage` ([6/PAYLOADS](/specs/6-payloads)) with a `payload` of `CommunityMessageArchiveIndex`.
+
+Only the community owner has permission to send messages with this topic.
+Community members MUST NOT have permission to send messages with this topic.
+However, community member nodes MUST subscribe to this topic to receive message archive indices.
+
+## Canonical message histories
+
+Only community owners are allowed to distribute messages with magnet links via the magnet link channel. Community members MUST NOT be allowed to distribute magnet links. Since the magnet links are created from the community owner node's database (and previously distributed archives), the message history provided by the community owner becomes the canonical message history and single source of truth for the community.
+
+Community member nodes MUST replace messages in their local databases with the messages extracted from archives within the same time range. Messages that didn't receive the community owner node MUST be removed and are no longer part of the message history of interest, even if it already existed in a community member node's database.
+
+
+## Fetching message archives
+
+Generally, fetching message archives is a tree step process:
+
+1. Receive message archive index signal, download index, then determine which message archives to download
+3. Download individual archives
+
+Community member nodes subscribe to the read-only channel that community owner nodes publish message archive indices to. There are two scenarios in which member nodes can receive message archive index messages:
+
+1. The member node receives it via live messages, that is, messages that are relayed by store nodes 
+2. The member node requests messages for a time range of up to 30 days from store nodes (this is the case when a new community member joins a community)
+
+### Downloading message archive indices
+When member nodes receive a message with a `CommunityMessageArchiveIndex` ([6/PAYLOADS](/specs/6-payloads)) from the aforementioned channnel, they MUST extract the `magnet_uri` and pass it to their underlying BitTorrent client so they can fetch the latest message archive index.
```

Comment:
I thought member nodes receive a waku message whose payload is a `ApplicationMetadataMessage` which embodies a `CommunityMessageArchiveIndex` as its payload.

Reply:
I've renamed this to `CommunityMessageArchive`.

But yes, that's the payload. And it has the `magnet_uri` that needs to be passed to the bittorrent client.